core/elements/ons-ripple.es6

```diff
@@ -0,0 +1,232 @@
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+(() => {
+  'use strict';
+
+  const scheme = {'': 'ripple--*'};
+  const ModifierUtil = ons._internal.ModifierUtil;
+
+  class RippleElement extends ons._BaseElement {
+
+    createdCallback() {
+      this.classList.add('ripple');
+      this._compile();
+      ModifierUtil.initModifier(this, scheme);
+    }
+
+    _compile() {
+      this._wave = document.createElement('span');
+      this._wave.classList.add('ripple__wave');
+      this.insertBefore(this._wave, this.children[0]);
+
+      if (this.hasAttribute('color')) {
+        this._wave.style.background = this.getAttribute('color');
+      }
+    }
+
+    _updateTarget() {
+      if (this.hasAttribute('target') && this.getAttribute('target') === 'children') {
+        this._target = this;
+        this.style.display = 'inline-block';
+        this.style.position = 'relative';
+      } else {
+        this._target = this.parentNode;
+        if (window.getComputedStyle(this._target).getPropertyValue('position') === 'static') {
+          this._target.style.position = 'relative';
+        }
+        this.style.display = 'block';
+        this.style.position = 'absolute';
+      }
+    }
+
+    _updateCenter() {
+      if (this.hasAttribute('center')) {
+        this._center = true;
+      } else {
+        this._center = false;
+      }
+      this._wave.classList.remove('ripple__wave--done-center');
+      this._wave.classList.remove('ripple__wave--animate-center');
+      this._wave.classList.remove('ripple__wave--done');
+      this._wave.classList.remove('ripple__wave--animate');
+    }
+
+    attributeChangedCallback(name, last, current) {
+      if (name === 'modifier') {
+        return ModifierUtil.onModifierChanged(last, current, this, scheme);
+      }
+      if (name === 'target') {
+        this._updateTarget();
+      }
+      if (name === 'color') {
+        this._wave.style.background = current;
+      }
+      if (name === 'center') {
+        this._updateCenter();
+      }
+      if (name === 'disabled') {
+        if (this.isDisabled()) {
+          if (this.querySelector('.ripple__wave')) {
+            this.removeChild(this.querySelector('.ripple__wave'));
+          }
+        } else {
+          this._compile();
+        }
+      }
+    }
+
+    _addListener() {
+      this.addEventListener('mousedown', this._onMouseDown.bind(this));
+      this.addEventListener('touchstart', this._onMouseDown.bind(this));
+    }
+
+    _removeListener() {
+      this.removeEventListener('mousedown', this._onMouseDown.bind(this));
```

Comment:
This doesn't work. You need to assign the event listener function to a variable in order to be able to remove it.